cpp/src/arrow/compute/exec/union_node_test.cc

```diff
@@ -17,187 +17,126 @@
-                     int num_batches, int batch_size) {
-  auto l_schema = schema({field("colum0", data_type), field("colum1", data_type)});
-  auto r_schema = schema({field("colum0", data_type), field("colum1", data_type)});
-
-  // generate data
-  auto l_batches = MakeRandomBatches(l_schema, num_batches, batch_size);
-  auto r_batches = MakeRandomBatches(r_schema, num_batches, batch_size);
-
-  ASSERT_OK_AND_ASSIGN(auto plan, ExecPlan::Make());
-
-  Declaration Union{"union", ExecNodeOptions{}};
-
-  // add left source
-  Union.inputs.emplace_back(Declaration{
-      "source", SourceNodeOptions{l_batches.schema, l_batches.gen(parallel,
-                                                                  /*slow=*/false)}});
-  // add right source
-  Union.inputs.emplace_back(Declaration{
-      "source", SourceNodeOptions{r_batches.schema, r_batches.gen(parallel,
-                                                                  /*slow=*/false)}});
-  AsyncGenerator<util::optional<ExecBatch>> sink_gen;
 
-  ASSERT_OK(Declaration::Sequence({Union, {"sink", SinkNodeOptions{&sink_gen}}})
-                .AddToPlan(plan.get()));
+  ::arrow::random::SeedType seed_ = 0xdeadbeef;
+  ::arrow::random::RandomArrayGenerator rng_;
 
-  auto actual = StartAndCollect(plan.get(), sink_gen);
+  static constexpr int kNumBatches = 10;
+  static constexpr int kBatchSize = 10;
+};
 
-  BatchesWithSchema exp_batches;
-  exp_batches.schema = l_schema;
-  exp_batches.batches.reserve(l_batches.batches.size() + r_batches.batches.size());
-
-  std::copy(l_batches.batches.begin(), l_batches.batches.end(),
-            std::back_inserter(exp_batches.batches));
-  std::copy(r_batches.batches.begin(), r_batches.batches.end(),
-            std::back_inserter(exp_batches.batches));
-
-  auto expected_matcher =
-      Finishes(ResultWith(UnorderedElementsAreArray(exp_batches.batches)));
-  ASSERT_THAT(actual, expected_matcher);
+TEST_F(TestUnionNode, TestNonEmpty) {
+  for (bool parallel : {false, true}) {
+    for (int64_t num_input_nodes : {1, 2, 4, 8}) {
+      this->CheckUnionExecNode(num_input_nodes, /*num_batches=*/kNumBatches, parallel);
+    }
+  }
 }
+TEST_F(TestUnionNode, TestWithAnEmptyBatch) { this->CheckUnionExecNode(2, 0, false); }
 
-class UnionTestRand
-    : public testing::TestWithParam<std::tuple<std::shared_ptr<DataType>, bool>> {};
-
-static constexpr int kNumBatches = 100;
-static constexpr int kBatchSize = 10;
-
-INSTANTIATE_TEST_SUITE_P(UnionTestRand, UnionTestRand,
-                         ::testing::Combine(::testing::Values(int8(), int32(), int64(),
-                                                              float32(), float64()),
-                                            ::testing::Values(false, true)));
-
-TEST_P(UnionTestRand, TestingTypes) {
-  TestUnionRandom(std::get<0>(GetParam()), std::get<1>(GetParam()), kNumBatches,
-                  kBatchSize);
-}
+TEST_F(TestUnionNode, TestEmpty) { this->CheckUnionExecNode(0, 0, false); }
```

Comment:
Nit: usually for constant parameters like this, we prefix the value with the argument name. See https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

Reply:
```suggestion
TEST_F(TestUnionNode, TestEmpty) {
  this->CheckUnionExecNode(/*num_input_nodes*/0, /*num_batches=*/0, /*parallel=*/false);
}
```

Reply:
And conversely above in TestNonEmpty, the comment is redundant since there's a constant.